src/registry/artifactory.rs

```diff
@@ -91,17 +106,29 @@ impl Registry for Artifactory {
             url
         };
 
-        let response = reqwest::Client::builder()
-            .redirect(reqwest::redirect::Policy::none())
-            .build()
-            .wrap_err("client error")?
-            .get(artifact_uri.clone())
-            .header(
-                "X-JFrog-Art-Api",
-                self.0.password.clone().unwrap_or_default(),
-            )
-            .send()
-            .await?;
+        let response = match self.credentials.get(&dependency.manifest.registry) {
+            Some(token) => {
+                // Go on with auth.
+                reqwest::Client::builder()
+                    .redirect(reqwest::redirect::Policy::none())
+                    .build()
+                    .wrap_err("client error")?
+                    .get((*artifact_uri).clone())
+                    .header("X-JFrog-Art-Api", &**token)
+                    .send()
+                    .await?
+            }
+            None => {
+                // Go on without auth.
+                reqwest::Client::builder()
+                    .redirect(reqwest::redirect::Policy::none())
+                    .build()
+                    .wrap_err("client error")?
+                    .get((*artifact_uri).clone())
+                    .send()
+                    .await?
+            }
+        };
```

Comment:
you could dedup code by making the builder mutable and only appending the header if you happen to have a token

Reply:
Agreed, that's on the horizon. I just copied for now because it allowed me to move faster.

Reply:
The header method consumes self, so it's not quite as pretty as declaring the builder mut, but it's okay..

```rust
pub fn header<K, V>(self, key: K, value: V) -> RequestBuilder
```

I can still do this:

```rust
        let response = {
            let builder = reqwest::Client::builder()
                .redirect(reqwest::redirect::Policy::none())
                .build()
                .wrap_err("client error")?
                .get(repositories_uri.as_str());

            let builder = if let Some(token) = self.credentials.credentials.get(&self.registry) {
                builder.header(JFROG_AUTH_HEADER, token)
            } else {
                builder
            };

            builder.send().await?
        };
```

Reply:
I think that is much more readable in terms of "what is the code doing"